src/app/components/Countdown/Countdown.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import styled from 'styled-components'
+
+type CountdownProps = {
+  deadline: number
+}
+
+export default function Countdown({ deadline }: CountdownProps): JSX.Element {
+  const diff = deadline - new Date().getTime()
+
+  const isDue = diff < 0
+  const yearDiff = Math.floor(diff / (1000 * 60 * 60 * 24 * 365))
+  const monthDiff = Math.floor(diff / (1000 * 60 * 60 * 24 * 30))
+  const dayDiff = Math.floor(diff / (1000 * 60 * 60 * 24))
+  const hourDiff = Math.floor(diff / (1000 * 60 * 60))
+
+  const countdownString = isDue
+    ? '!'
+    : yearDiff
+    ? `${yearDiff}Y`
+    : monthDiff
+    ? `${monthDiff}M`
+    : dayDiff
+    ? `${dayDiff}D`
+    : `${hourDiff}H`
```

Comment:
I don't see why you'd need to chain all these ternaries, will the variables other than `diff` ever be falsy?